websites/D/Desu-Online/presence.ts

```diff
@@ -0,0 +1,166 @@
+const presence = new Presence({
+		clientId: "580032576434077707",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+let video = {
+	current: 0,
+	duration: 0,
+	paused: true,
+};
+
+presence.on(
+	"iFrameData",
+	(data: { current: number; duration: number; paused: boolean }) => {
+		video = data;
+	}
+);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "logo",
+			startTimestamp: browsingTimestamp,
+		},
+		pth = document.location.pathname;
+	if (document.location.host === "desu-online.pl") {
+		presenceData.smallImageKey =
+			"https://cdn-icons-png.flaticon.com/512/5611/5611971.png";
+		if (document.location.search !== "") {
+			presenceData.details = "Używa zaawansowanej wyszukiwarki...";
+			const searchvalues = document.location.search.split("&");
+			for (const searchvalue of searchvalues) {
+				if (searchvalue.startsWith("s=")) {
+					presenceData.details = "Wyszukuje po frazie:";
+					presenceData.state = searchvalue
+						.substring(2, searchvalue.length)
+						.replaceAll("+", " ")
+						.replaceAll("%20", " ");
+				} else if (searchvalue.startsWith("?s=")) {
+					presenceData.details = "Wyszukuje po frazie:";
+					presenceData.state = searchvalue
+						.substring(3, searchvalue.length)
+						.replaceAll("+", " ")
+						.replaceAll("%20", " ");
+				}
+			}
+		} else if (pth === "/") {
+			presenceData.details = "Przegląda stronę główną...";
+			presenceData.smallImageKey =
+				"https://cdn-icons-png.flaticon.com/512/5974/5974636.png";
+		} else if (pth.startsWith("/anime/")) {
+			if (pth === "/anime/") presenceData.details = "Przegląda serie...";
+			else if (pth === "/anime/list-mode/")
+				presenceData.details = "Przegląda listę serii...";
+			else {
+				presenceData.details = "Przegląda serię:";
+				presenceData.state = document.querySelectorAll("h1")[0].textContent;
+				presenceData.largeImageKey = document
+					.querySelector("img.ts-post-image")
+					.getAttribute("src");
+				presenceData.buttons = [
+					{
+						label: "Odwiedź",
+						url: document.URL,
+					},
+				];
+			}
+		} else if (pth === "/harmonogram/")
+			presenceData.details = "Przegląda harmonogram...";
+		else if (pth === "/zakladki/")
+			presenceData.details = "Przegląda swoje zakładki...";
+		else if (pth.startsWith("/blog/")) {
+			if (pth === "/blog/") {
+				presenceData.details = "Przegląda bloga:";
+				presenceData.state = "Strona 1";
+			} else if (pth.startsWith("/blog/page/")) {
+				presenceData.details = "Przegląda bloga:";
+				presenceData.state = `Strona ${
+					document.location.pathname.split("/")[3]
```

Comment:
You already have a variable for `document.location.pathname`.
```suggestion
					pth.split("/")[3]
```